sw/airborne/firmwares/rotorcraft/stabilization/stabilization_indi.c

```diff
@@ -340,45 +363,52 @@ static void stabilization_indi_calc_cmd(struct Int32Quat *att_err, bool rate_con
   //G2 is scaled by INDI_G_SCALING to make it readable
   g2_times_du = g2_times_du/INDI_G_SCALING;
 
-  // Calculate the increment for each actuator
-  for(i=0; i<INDI_NUM_ACT; i++) {
-    indi_du[i] = (g1g2_pseudo_inv[i][0] * (angular_accel_ref.p - angular_acceleration[0]))
-               + (g1g2_pseudo_inv[i][1] * (angular_accel_ref.q - angular_acceleration[1]))
-               + (g1g2_pseudo_inv[i][2] * (angular_accel_ref.r - angular_acceleration[2] + g2_times_du));
-  }
-
+  float v_thrust = 0.0;
   if(indi_thrust_increment_set){
-    // The required body-z acceleration is calculated by the outer loop INDI controller
-    for(i=0; i<INDI_NUM_ACT; i++) {
-      indi_du[i] = indi_du[i] + g1g2_pseudo_inv[i][3]*indi_thrust_increment;
-    }
+    v_thrust = indi_thrust_increment;
 
-    // Add the increments to the actuators
-    float_vect_sum(indi_u, actuator_state_filt_vect, indi_du, INDI_NUM_ACT);
-  } else
-  {
-    // Add the increments to the actuators without the thrust
-    float_vect_sum(indi_u, actuator_state_filt_vect, indi_du, INDI_NUM_ACT);
-
-    // Calculate the average of the actuators as a measure for the thrust
-    float avg_u_in = 0;
+    //update thrust command such that the current is correctly estimated
+    stabilization_cmd[COMMAND_THRUST] = (actuator_state[0] + actuator_state[1] + actuator_state[2] + actuator_state[3])/4.0;
+  } else {
+    // incremental thrust
     for(i=0; i<INDI_NUM_ACT; i++) {
-      avg_u_in += indi_u[i];
+      v_thrust +=
+        (stabilization_cmd[COMMAND_THRUST] - actuator_state_filt_vect[i])*Bwls[3][i];
     }
-    avg_u_in /= INDI_NUM_ACT;
+  }
 
-    // Make sure the thrust is bounded
-    Bound(stabilization_cmd[COMMAND_THRUST],0, MAX_PPRZ);
+  // Calculate the min and max increments
+  for(i=0; i<INDI_NUM_ACT; i++) {
+    du_min[i] = -MAX_PPRZ*act_is_servo[i] - actuator_state_filt_vect[i];
+    du_max[i] = MAX_PPRZ - actuator_state_filt_vect[i];
+    du_pref[i] = act_pref[i] - actuator_state_filt_vect[i];
+  }
 
-    //avoid dividing by zero
-    if(avg_u_in < 1.0) {
-      avg_u_in = 1.0;
-    }
+  //State prioritization {W Roll, W pitch, W yaw, TOTAL THRUST}
+  static float Wv[INDI_OUTPUTS] = {1000, 1000, 1, 100};
+
+  // The control objective in array format
+  indi_v[0] = (angular_accel_ref.p - angular_acceleration[0]);
```

Comment:
They are not linked, I have just fixed this for the user. I think this is what 99% of people will need, so maybe making it very configurable at this stage does not make too much sense.